src/OpenTelemetry.Instrumentation.Http/HttpClientInstrumentationOptions.cs

```diff
@@ -157,6 +138,11 @@ internal HttpClientInstrumentationOptions(IConfiguration configuration)
     /// </remarks>
     public bool RecordException { get; set; }
 
+    /// <summary>
+    /// Gets or sets a value indicating whether suppresses http instrumentation when the http call is part of grpc client call.
+    /// </summary>
+    public bool SuppressInstrumentationWhenGrpcIsPresent { get; set; }
```

Comment:
I think we should ship this new option as experimental. The option is useful as a stop gap, but I do not think it's ultimately what we want. We have an open issue with a different, more generic, proposal #4641 though it is only focused on suppressing tracing. The nice thing about our current suppression functionality is that it works to suppress all signals.

The ability to suppress downstream instrumentation is a long running [open issue](https://github.com/open-telemetry/opentelemetry-specification/issues/530) in the specification. Ultimately, I believe this will be functionality provided by the API not the SDK.

I am ok with this PR if we remove the dependency on the SDK from the HTTP instrumentation and ship this option specifically targeting gRPC as experimental.

Reply:
I removed the option proposed for now. Will bring it back after the stable release. Was hoping to use `EXPOSE_EXPERIMENTAL_FEATURES` but it is not possible to use it as is without additional changes in build. We can discuss the possible options (the one I proposed and existing one later).